 URL : "https://github.com/rust-lang-nursery/error-chain/issues/219" TITLE : how to chain nom::ierror? BODY : rust foreign_links { nom ::nom::ierror ; io ::std::io::error ; parseint ::std::num::parseinterror ; } error e0599 : no method named cause found for type &nom::ierror in the current scope --> src/main.rs:56:5 | 56 | / error_chain! { 57 | | foreign_links { 58 | | nom ::nom::ierror ; 59 | | io ::std::io::error ; ... | 65 | | } 66 | | } | |_____^ | = note: this error originates in a macro outside of the current crate error e0277 : the trait bound nom::ierror: std::fmt::display is not satisfied --> src/main.rs:56:5 | 56 | / error_chain! { 57 | | foreign_links { 58 | | nom ::nom::ierror ; 59 | | io ::std::io::error ; ... | 65 | | } 66 | | } | |_____^ nom::ierror cannot be formatted with the default formatter; try using :? instead if you are using a format string | = help: the trait std::fmt::display is not implemented for nom::ierror = note: required because of the requirements on the impl of std::fmt::display for &nom::ierror = note: required by std::fmt::display::fmt = note: this error originates in a macro outside of the current crate error e0277 : the trait bound nom::ierror: std::error::error is not satisfied --> src/main.rs:56:5 | 56 | / error_chain! { 57 | | foreign_links { 58 | | nom ::nom::ierror ; 59 | | io ::std::io::error ; ... | 65 | | } 66 | | } | |_____^ the trait std::error::error is not implemented for nom::ierror | = note: required by std::error::error::description = note: this error originates in a macro outside of the current crate error: aborting due to 3 previous errors